### PR TITLE
:whale: deploy with Docker Compose

### PR DESCRIPTION
**Pull request by nomaster**
I have made some modifications to deploy mete with `docker-compose`. The HTTP service is always exposed on port 80, so we should make use of a dedicated VM.

Features:
 - move from sqlite to PostgreSQL (for more stability and scalability)
 - switch to Puma webserver (speedier and more recent)
 - backup service that creates database dumps on an hourly basis (very basic implementation)
 - automatic database creation and initialization (see docker-entrypoint.sh)
 
Bugs/Limitations:
 - users and drinks are no longer sorted 'nocase' (PostgreSQL doesn't support it that way) 

Future ideas:
 - create backups from assets as well send backups

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,6 +15,10 @@ gem 'sass-rails'
 gem 'coffee-rails'
 gem 'bootstrap-sass', :github => 'thomas-mcdonald/bootstrap-sass'
 
+# See https://github.com/sstephenson/execjs#readme for more supported runtimes
+gem 'therubyracer', :platforms => :ruby
```
Comment (marudor) [inline Gemfile]: rubyracer haben wir eig. absichtlich mal rausgehauen.
Der macht oft Probleme beim bauen.
Alternative war node installieren und execjs - dann wird node als javascript runtime genutzt.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    environment:
+      PGPASSWORD: mete
+    volumes:
+      - backup:/backup
+    restart: on-failure
+    networks:
+      appservers:
+
+  backend:
+    image: nomaster/mete
```
Comment (marudor) [inline docker-compose.yml]: könnten wir das aufs chaosdorf mappen statt auf dich privat?

File: Gemfile (hunk 1 above)
Comment (nomaster) [reply Gemfile]: Ich kann mich nicht erinnern, den hinzugefügt zu haben. 🤔 Sorry, hau ich wieder raus.

File: docker-compose.yml (hunk 2 above)
Comment (nomaster) [reply docker-compose.yml]: Ja klar. Henne und Ei Problem an dieser Stelle noch, weil das Image ja aus einem anderen Repo kommt. Aber sollte egal sein. Wird gemacht!

**Comment by YtvwlD** [comment]
I just took a quick look, but this looks a bit excessive.

A few notes:
 * Do we really need a separate VM? Couldn't we use a single webserver for HTTPS termination and such for multiple services (Labello, Dashboard, ...)?
 * What are the advantages of using Puma?
 * What are the advantages of using Postgres?
 * The sorting is important for the web ui ad various clients. We need a good solution there.
 * We are not the only ones using this.

**Comment by nomaster** [comment]
Separate VM because we don't have a proper container cluster running yet.  This means port binding and proxying must be taken care of manually.

Puma is the new default server for Rails and is said to deliver more performance.

PostgreSQL ist a proper DB service and makes concurrent access possible. This is important for consistent dumping (backups) and makes scaling of the backend possible.

The sorting should be implemented with a slug name property and a separate column. This way it's faster and better probable. Raw SQL should be avoided IMHO.

**Comment by YtvwlD** [comment]
So, using Postgres seems sensible.
I don't really know about sorting - adding another column might not be the way to go. [Case-insensitive sorting seems to work with Postgres, too.](https://stackoverflow.com/questions/2983687/how-to-do-case-insensitive-order-in-rails-with-postgresql)

If Puma is the default server, let's just use it.

**Pull request merged**